pkg/helios/cas/cas.go

```diff
@@ -0,0 +1,398 @@
+// Based on research findings showing BLAKE3's superior performance characteristics:
+// - ~15 GB/s throughput with AVX2
+// - <100ns latency for small inputs  
+// - SIMD acceleration support
+type BLAKE3Store struct {
+	storePath   string
+	cache       map[string][]byte // L1 cache for hot content
+	mutex       sync.RWMutex      // Protects cache for concurrent access
+	
+	// Performance optimizations from research phase
+	hasherPool  sync.Pool         // Pool of reusable BLAKE3 hashers
+	keyCache    map[string]string // Pre-computed hex keys for hot paths
+	keyMutex    sync.RWMutex      // Protects key cache
+	
+	// Ultra-performance optimizations for <70μs VST targets
+	memoryMode  bool              // Skip disk I/O for maximum performance
+	writeQueue  chan writeOp      // Async write queue for background persistence
+	errorQueue  chan error        // Channel for background write errors
+	wg          sync.WaitGroup    // Wait group for background writes
+	closed      bool              // Track if store is closed
+	closeMutex  sync.Mutex        // Protects close operations
+}
+
+// NewBLAKE3Store creates a new BLAKE3-based content-addressable store
+// storePath: directory for persistent storage
+func NewBLAKE3Store(storePath string) (*BLAKE3Store, error) {
+	// Ensure storage directory exists
+	if err := os.MkdirAll(storePath, 0755); err != nil {
+		return nil, fmt.Errorf("failed to create storage directory: %w", err)
+	}
+
+	store := &BLAKE3Store{
+		storePath:  storePath,
+		cache:      make(map[string][]byte),
+		keyCache:   make(map[string]string),
+		memoryMode: false, // Default to persistent mode
+		writeQueue: make(chan writeOp, 1000), // Buffered channel for async writes
+		errorQueue: make(chan error, 100),    // Buffered channel for errors
+	}
+	
+	// Initialize hasher pool for zero-allocation hashing
+	store.hasherPool = sync.Pool{
+		New: func() interface{} {
+			return blake3.New(32, nil) // Pre-configured 256-bit unkeyed hasher
+		},
+	}
+	
+	// Start background writer goroutine
+	go store.backgroundWriter()
+	
+	// Start error handler goroutine
+	go store.errorHandler()
+	
+	return store, nil
+}
+
+// backgroundWriter handles async disk writes for maximum performance
+func (s *BLAKE3Store) backgroundWriter() {
+	for writeOp := range s.writeQueue {
+		if err := os.WriteFile(writeOp.filePath, writeOp.content, 0644); err != nil {
+			// Send error to error handler for logging/retry
+			select {
+			case s.errorQueue <- fmt.Errorf("background write failed for %s: %w", writeOp.filePath, err):
+			default:
+				// Error queue full, error will be dropped but write continues
+			}
+		}
+		s.wg.Done()
+	}
+}
+
+// errorHandler processes background write errors
+func (s *BLAKE3Store) errorHandler() {
+	for err := range s.errorQueue {
+		// Log error to stderr for proper error handling
+		// Using fmt.Fprintf to stderr instead of stdout for errors
+		fmt.Fprintf(os.Stderr, "[ERROR] BLAKE3Store background write failed: %v\n", err)
```

Comment:
Library ergonomics: printing to stderr from a package can be surprising. Consider injecting a logger interface or callback so hosts can control logging, or guard with a debug flag.